lib/wpscan/web_site.rb

```diff
@@ -55,6 +55,9 @@ def redirection(url = nil)
     redirected_uri = URI.parse(add_trailing_slash(add_http_protocol(url)))
     if response.code == 301 || response.code == 302
       redirection = response.headers_hash['location']
+      if redirection[0] == '?'
+        redirection = "/#{redirection}"
```

Comment:
You could even take advantage of the URI#merge method to replace those two if:

```
2.2.1 :001 > require 'URI'
 => true 
2.2.1 :002 > uri = URI.parse 'https://g.com/dir/'
 => #<URI::HTTPS https://g.com/dir/> 
2.2.1 :003 > uri.merge('/root')
 => #<URI::HTTPS https://g.com/root> 
2.2.1 :004 > uri.merge('?param=1')
 => #<URI::HTTPS https://g.com/dir/?param=1> 
2.2.1 :005 > uri.merge('//h.org')
 => #<URI::HTTPS https://h.org> 
2.2.1 :006 > uri.merge('http://i.net')
 => #<URI::HTTP http://i.net>
```

Btw, in your case, if the redirected_ui is a subdir (e.g http://g.com/dir/) and the redirect something like ?p=2, the redirection returned will be https://g.com/?p=2 rather than the correct one. Using the URI#merge mentioned above will take care of such case ;)